roles/serviceassurance/tasks/_local_signing_authority.yml

```diff
@@ -7,77 +7,32 @@
       apiVersion: certmanager.k8s.io/v1alpha1
       kind: Issuer
       metadata:
-        name: '{{ meta.namespace }}-selfsigned'
+        name: '{{ meta.name }}-selfsigned'
         namespace: '{{ meta.namespace }}'
       spec:
         selfSigned: {}
 
-- name: Create CA certificate
-  k8s:
-    definition:
+- name: Create default Certificate manifest for Elasticsearch
+  set_fact:
+    elasticsearch_certificate_manifest: |
       apiVersion: certmanager.k8s.io/v1alpha1
       kind: Certificate
       metadata:
-        name: '{{ meta.namespace }}-ca'
-        namespace: '{{ meta.namespace }}'
+        name: elasticsearch-es-cert
+        namespace: default
       spec:
-        commonName: '{{ meta.namespace }}-ca'
+        dnsNames:
+          - elasticsearch-es-http
+          - elasticsearch-es-http.{{ meta.namespace }}.svc
+          - elasticsearch-es-http.{{ meta.namespace }}.svc.cluster.local
         isCA: true
```

Comment:
This looks wrong to me. You are saying that the cert that ES presents is a CA certificate (meant for signing other certificates).

Here is my understanding:

The original code created a CA certificate using the "self-signed" Issuer. From that a "namespace CA Issuer" was created, which is responsible for issuing and signing other certificates in this namespace. The ES certificates were then issued and signed by that namespace CA Issuer. This is a good model of how SSL works in the wild; and the benefit is that if you import and trust the CA certificate, you do not need to individually import and trust every certificate that it issues.

After your change, there is a single self-signed certificate for ES, which is probably okay, but the downside of the new model is that either:
1. Each new cert we ever create needs to be self-signed and imported individually into peers that might need to trust them (since there is no general-purpose CA we can trust)
OR
2. Every new cert we create will be signed by a cert called "elasticsearch-es-cert" which will likely lead to confusion

Neither of these is the end of the world, but I'd like to understand why you are making the change and what we're getting in return.

Reply:
Honestly, I blindly followed the samples provided by the documentation:

https://www.elastic.co/guide/en/cloud-on-k8s/1.0/k8s-custom-http-certificate.html#k8s_custom_self_signed_certificate_using_cert_manager

Reply:
All good points though, and I think it's worth trying to put the existing functionality back in and building up the certificates with the sign-signing CA cert as you describe. Let me look at it and try testing that setup.

Reply:
One last thing, was I was looking at the docs trying to understand if ECK maybe just used the CA cert and then self-generated certificates that get exposed using that, but I don't see anything on that page that says that's how it works.

Reply:
The new code is certainly simpler and easier to understand, but the ability to distribute just 1 cert CA cert to trust for all purposes seems worth maintaining the extra complexity here.

Reply:
Agreed. Thanks for stepping through that. I'll re-implement the logic to build a CA signing cert and then create certs from that for consumption by ES.